nuitka/optimizations/OptimizeBuiltinCalls.py

```diff
@@ -1000,6 +1015,188 @@ def wrapExpressionBuiltinExecCreation(
+        assign_iter_statements = [
+            StatementAssignmentVariable(
+                variable=zip_iter_variable,
+                source=ExpressionBuiltinIter1(
+                    value=ExpressionTempVariableRef(
+                        variable=zip_arg_variable, source_ref=source_ref
+                    ),
+                    source_ref=source_ref,
+                ),
+                source_ref=source_ref,
+            )
+            for zip_iter_variable, zip_arg_variable in zip(
+                zip_iter_variables, zip_arg_variables
+            )
+        ]
+
+        # makeTryExceptSingleHandlerNode
+        statements += [
+            makeTryExceptSingleHandlerNode(
+                tried=assign_iter_statements,
+                exception_name="TypeError",
+                handler_body=makeRaiseExceptionExpressionFromTemplate(
+                    exception_type="TypeError",
+                    template="zip argument #%s must support iteration",
+                    template_args=makeConstantRefNode(
+                        constant=str(i + 1), source_ref=source_ref
+                    ),
+                    source_ref=source_ref,
+                ),
+                source_ref=source_ref,
+            )
+            for i, _ in enumerate(zip_arg_variables)
+        ]
+
+        tmp_result = outline_body.allocateTempVariable(
+            temp_scope=None, name="tmp_result"
+        )
+
+        # When we append all values at once, all we
+        # need to do is to check StopIteration for
+        # any of the next attempts, ExpressionMakeTuple
+        # takes care of all releasing.
+        append_statement = StatementListOperationAppend(
+            list_arg=ExpressionTempVariableRef(
+                variable=tmp_result, source_ref=source_ref
+            ),
+            value=ExpressionMakeTuple(
+                elements=[
+                    ExpressionBuiltinNext1(
+                        value=ExpressionTempVariableRef(
+                            variable=zip_iter_variable, source_ref=source_ref
+                        ),
+                        source_ref=source_ref,
+                    )
+                    for zip_iter_variable in zip_iter_variables
+                ],
+                source_ref=source_ref,
+            ),
+            source_ref=source_ref,
+        )
+
+        # Can check all next in once go.
+        loop_body = makeStatementsSequenceFromStatement(
+            statement=makeTryExceptSingleHandlerNode(
+                tried=append_statement,
+                exception_name="StopIteration",
+                handler_body=StatementLoopBreak(source_ref=source_ref),
+                source_ref=source_ref,
+            )
+        )
+
+        statements.append(StatementLoop(body=loop_body, source_ref=source_ref))
+
+        statements.append(
+            StatementReturn(
+                ExpressionTempVariableRef(
+                    # TODO: Create that tmp_result
```

Comment:
Without solving this TODO, it will crash (trigger an assertion with --`debug`) at run time, because the temp variable won't be assigned.

Reply:
I think I have resolved this TODO in Line 1113. But it's still crashing

Reply:
And I'm getting `core dump`

Reply:
Allocating a variable, merely creates an object, it does nothing with it. You need to assign a value to that variable for it to be assigned. I am not referring to `tmp_result` as a Python level variable in your re-formulation. I am referring to it as the run time temporary variable of the outline, which without a `StatementAssignVariable` is going to be very crash happy if referenced before or without it.